### PR TITLE
Quiz Host Final Standings

### LINKED ISSUE #158: As a QuizMaster i want to see the questions summary with the final standings
All the participants get to see a questions summary page at the end of the quiz below the final standings, as a quiz master i don't see this,  only the final standings 
As a quiz master i would like to see this questions summary so that I can join in with the fun and the post match analysis of the questions and fastest answerers. 

The quiz master's summary would be slightly different since they themselves have not answered questions so they would not see "your answer" and would not get colour highlighting based on being correct or fastest

### PR DESCRIPTION
**Pull request by timloudon**
Closes #158

### CODE REVIEW

File: quizmaster-client-app/src/components/QuizHosting/QuizPlaying/QuizHoster.tsx [hunk 1]
```diff
@@ -1,580 +1,583 @@
...
+                />
+              </>
+            ) : (
+              <div className={classes.finalStandings}>
+                <h1>Final Standings</h1>
+                <QuizStandings
+                  contestantStandings={contestants}
+                  quizState={currentQuizState}
+                />
+                <h2>Questions</h2>
```
Comment (timloudon) [inline quizmaster-client-app/src/components/QuizHosting/QuizPlaying/QuizHoster.tsx:565]: The changes to this file start here (lines 565 & 566)

**Comment by timloudon** [comment]
QuizHoster file shows every line as a change because of merging Rich's updates into my out of date local branch. There's a note in there showing the two lines I've changed

File: QuizMaster.Api/Controllers/QuizzesController.cs [hunk 2]
```diff
@@ -79,6 +80,20 @@ public async Task<ActionResult<Details.QuizMasterQuizDetails>> GetQuizMasterQuiz
...
+            var quizSummary = await mediator.Send(new QuizMaster.Application.Quizzes.FinalSummary.Query(id));
+
+            if (quizSummary == null)
+            {
+                return NotFound();
+            }
+            return Ok(quizSummary);
+        }
+
+
```
Comment (calumtomeny) [inline QuizMaster.Api/Controllers/QuizzesController.cs:96]:
```suggestion
```

**Comment by timloudon** [comment]
CRLF issue resolved now

File: QuizMaster.Api/QuizMaster.Api.csproj [hunk 3]
```diff
@@ -17,5 +17,6 @@
   <PropertyGroup>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <DebugType>Portable</DebugType>
```
Comment (calumtomeny) [inline QuizMaster.Api/QuizMaster.Api.csproj:20]: Do we still need this?

File: QuizMaster.Application/Quizzes/FinalSummary.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                    questionSummaries.Add(
+                        new QuestionSummary
+                        {
+                            Number = question.Number,
+                            Question = question.Question,
+                            Answer = question.Answer,
+                            Contestants = answers,
+                        }
+                    );
+
```
Comment (calumtomeny) [inline QuizMaster.Application/Quizzes/FinalSummary.cs:80]:
```suggestion
```

Comment (calumtomeny) [inline QuizMaster.Application/Quizzes/FinalSummary.cs:80]:
```suggestion
```

File: QuizMaster.Application/Quizzes/FinalSummary.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        public class Handler : IRequestHandler<Query, List<QuestionSummary>>
+        {
+            private readonly QuizContext context;
+            public Handler(QuizContext context)
+            {
+                this.context = context;
+            }
+            public async Task<List<QuestionSummary>> Handle(Query request, CancellationToken cancellationToken)
+            {
+                Quiz quiz = await context.Quiz.Include(x => x.Contestants).Include(x => x.QuizQuestions).ThenInclude(x => x.ContestantAnswers).SingleOrDefaultAsync(x => x.Code == request.QuizCode);
```
Comment (calumtomeny) [inline QuizMaster.Application/Quizzes/FinalSummary.cs:49]:
```suggestion
                Quiz quiz = await context.Quiz
                .Include(x => x.Contestants).Include(x => x.QuizQuestions).ThenInclude(x => x.ContestantAnswers)
                .SingleOrDefaultAsync(x => x.Code == request.QuizCode);
```

File: quizmaster-client-app/src/components/QuizHosting/QuizPlaying/QuizHoster.tsx [hunk 6]
```diff
@@ -480,11 +481,11 @@ export default function QuizHoster() {
...
-  //---------------------------------------------------------------------------
-  //---------------------------------------------------------------------------
-  //----------------------------------RENDER-----------------------------------
-  //---------------------------------------------------------------------------
-  //---------------------------------------------------------------------------
+  //----------------------------------------------------------------------------
+  //----------------------------------------------------------------------------
+  //----------------------------------RENDER------------------------------------
+  //----------------------------------------------------------------------------
+  //----------------------------------------------------------------------------
```
Comment (calumtomeny) [inline quizmaster-client-app/src/components/QuizHosting/QuizPlaying/QuizHoster.tsx:488]:
```suggestion
  //---------------------------------------------------------------------------
  //---------------------------------------------------------------------------
  //----------------------------------RENDER-----------------------------------
  //---------------------------------------------------------------------------
  //---------------------------------------------------------------------------
```

File: QuizMaster.Api/QuizMaster.Api.csproj (hunk 3 above)
Comment (timloudon) [reply QuizMaster.Api/QuizMaster.Api.csproj:20]: Probably not

**Review by richielewis — APPROVED** [review]
Looks good to me but have a look at Cal's comments.

**Pull request merged**